src/water_cmd.cpp

```diff
@@ -1237,7 +1240,7 @@ void TileLoop_Water(TileIndex tile)
 			uint dir;
 			FOR_EACH_SET_BIT(dir, _flood_from_dirs[slope_here]) {
 				TileIndex dest = tile + TileOffsByDir((Direction)dir);
-				if (!IsValidTile(dest)) continue;
+				if (dest >= MapSize()) continue;
```

Comment:
This change here is to allow the check on void tiles, which are now `FLOOD_ACTIVE`. We don't want `tile` to dry up in this situation.

Reply:
Can't add an optional `allow_void` bool to `IsValidTile()` ? (just a general question to everyone, not a suggestion for you)
I think it could simplify some tests in other parts of the code (based on my quick search for `>= MapSize()`.